security/certora/specs/Governance.spec

```diff
@@ -135,15 +136,33 @@ rule consecutiveIDs(method f) filtered
 
 // @title Property #2: Every proposal should contain at least one payload.
 // For initialized property, the proposal list is not empty
+/* ND:  since this is env dependent we are not checking:
+  require getProposalState(e, proposalId) != IGovernanceCore.State.Null  
+   => getPayloadLength(proposalId) > 0
+  f(e2,args)
+  assert require getProposalState(e3, proposalId) != IGovernanceCore.State.Null  
+   => getPayloadLength(proposalId) > 0
+
+  Can this be an issue? 
+  usually time dependnet properties should be proven as a rule 
+*/
+
 invariant at_least_single_payload_active (env e, uint256 proposalId)
-  getProposalState(e, proposalId) != IGovernanceCore.State.Null 
+  getProposalState(e, proposalId) != IGovernanceCore.State.Null  
    => getPayloadLength(proposalId) > 0
   {
       preserved{
       requireInvariant empty_payloads_if_uninitialized_proposal(proposalId);
     }
   }
 // Same property just referring directly to the storage
+/*
+ND: since getProposalStateVariable is a certora harneess function we need a rule checking 
+
+getProposalStateVariable(proposalId) != getProposalState(e,proposalId)
+
+I see many of our rules are on getProposalState() and some are also on getProposalStateVariable() but when and why? 
```

Comment:
We really need getProposalState() - that's the contract function that retrieves the state. All contract functions use this function.

getProposalStateVariable() is a getter to the storage. It's an extra check.